webapp/controller/PageOverview.controller.js

```diff
@@ -0,0 +1,199 @@
+	"use strict";
+
+	return Controller.extend("products.app.controller.PageOverview", {
+		
+		onInit: function() {
+			const oModel = new JSONModel();
+			oModel.loadData("../model/products.json");
+		
+			this.getView().setModel(oModel);
+		},
+
+		onAfterRendering: function() {
+			const oMultiInput = this.byId("multiInput");
+			oMultiInput.attachTokenUpdate((event) => {
+				if (event.getParameter("type") === "removed") {
+					const oModel = this.getView().getModel();
+					const aSuppliers = oModel.getProperty("/Suppliers");
+					
+					const aRemovedTokens = event.getParameter("removedTokens");
+				
+					const aRemainingTokens = oMultiInput.getTokens().filter(function (token) {
+					  return !aRemovedTokens.includes(token);
+					});
+					
+					const aSelectedTokens = aRemainingTokens.map(el => el.getText());
+					const aSelectedId = aSuppliers
+											.filter((el) => aSelectedTokens.includes(el.SuppliersName))
+											.map((el) => el.SupplierId);
+					
+					this.onFilter(aSelectedId);
+				}
+			})
+		},
+
+		getSuppliersName: function(data) {
+			const oModel = this.getView().getModel();
+			const aSuppliers = oModel.getProperty("/Suppliers");			
+			
+			return aSuppliers
+					.filter((el) => data.includes(el.SupplierId))
+					.map(el => el.SuppliersName)
+					.join(", ");
+		},
+
+		getCombinedFilter: function(aSelectedId) {
+			const sQuerySearch = this.byId("searcher").getValue();
+			const sSelectedKey = this.byId("availabilitySelect").getSelectedKey();
+			
+			const sAvailability = "all";
+			const oAvailabilityFilter = new Filter(
+				"Availability", 
+				FilterOperator.NE,
+				sSelectedKey === sAvailability ? sSelectedKey : JSON.parse(sSelectedKey.toLowerCase())
+			)
+			const aSearchFilters = [
+				new Filter("Name", FilterOperator.Contains, sQuerySearch),
+				new Filter("Descrition", FilterOperator.Contains, sQuerySearch)
+			]
+			const oSearchFilter = new Filter({
+				filters: aSearchFilters,
+				and: false
+			});
+			const oCategoryFilters = new Filter({
+				filters: this.getCategoriesFilters(),
+				and: false
+			})
+			
+			const oSuppliersFilters = new Filter({
+				filters: this.getSuppliersFilters(aSelectedId),
+				and: true
+			})
+			return new Filter({
+				filters: [oSearchFilter, oAvailabilityFilter, oCategoryFilters, oSuppliersFilters],
+				and: true
+			});
+		},
+
+		onFilter: function(aSelectedId = null) {
+			const oTable = this.byId("table");
+			const oItemsBinding = oTable.getBinding("items");
+			
+			const oFilter = this.getCombinedFilter(aSelectedId);
+			
+			oItemsBinding.filter(oFilter);
+		},
+
+		getSuppliersFilters: function(aSelectedId) {
+			const oMultiInput = this.byId("multiInput");
+			const oModel = this.getView().getModel();
+			const aAllSuppliers = oModel.getProperty("/Suppliers");
+			const aSuppliersItems = oMultiInput.getTokens().map(el => el.getText());
+			const aFilters = [];
+			let aSuppliersId = null;
+		
+			if(aSelectedId) {
```

Comment:
The condition is not quite correct. As aSelectedId is of type array, it will be always true